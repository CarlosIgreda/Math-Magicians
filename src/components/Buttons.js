function Buttons() {
  return (
    <div className="buttons">
      <div className="display">
        0
      </div>
      <button className="button numbers-and-extras" type="button">AC</button>
      <button className="button numbers-and-extras" type="button">+/-</button>
      <button className="button numbers-and-extras" type="button">%</button>
      <button className="button math-symbols" type="button">รท</button>
      <button className="button numbers-and-extras" type="button">7</button>
      <button className="button numbers-and-extras" type="button">8</button>
      <button className="button numbers-and-extras" type="button">9</button>
      <button className="button math-symbols" type="button">ร</button>
      <button className="button numbers-and-extras" type="button">4</button>
      <button className="button numbers-and-extras" type="button">5</button>
      <button className="button numbers-and-extras" type="button">6</button>
      <button className="button math-symbols" type="button">-</button>
      <button className="button numbers-and-extras" type="button">1</button>
      <button className="button numbers-and-extras" type="button">2</button>
      <button className="button numbers-and-extras" type="button">3</button>
      <button className="button math-symbols" type="button">+</button>
      <button className="button numbers-and-extras number0" type="button">0</button>
      <button className="button numbers-and-extras" type="button">.</button>
      <button className="button math-symbols" type="button">=</button>
    </div>
  );
}
export default Buttons;
