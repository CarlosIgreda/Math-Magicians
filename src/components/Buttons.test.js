import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Buttons from './Buttons';

describe('Buttons', () => {
  it('should render correctly', () => {
    const { container } = render(<Buttons />);
    expect(container).toMatchSnapshot();
  });

  it('should handle button clicks and update the display value', () => {
    const { getByText } = render(<Buttons />);

    // Click on buttons
    fireEvent.click(getByText('7'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('3'));
    fireEvent.click(getByText('='));

    // Check if display value is updated correctly
    expect(getByText('10')).toBeInTheDocument();
  });
});
