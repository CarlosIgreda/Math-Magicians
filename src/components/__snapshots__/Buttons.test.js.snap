// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should render correctly 1`] = `
<div>
  <div
    class="buttons"
  >
    <div
      class="display"
    >
      0
    </div>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      AC
    </button>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      +/-
    </button>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      %
    </button>
    <button
      class="button math-symbols"
      type="button"
    >
      รท
    </button>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      7
    </button>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      8
    </button>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      9
    </button>
    <button
      class="button math-symbols"
      type="button"
    >
      x
    </button>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      4
    </button>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      5
    </button>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      6
    </button>
    <button
      class="button math-symbols"
      type="button"
    >
      -
    </button>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      1
    </button>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      2
    </button>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      3
    </button>
    <button
      class="button math-symbols"
      type="button"
    >
      +
    </button>
    <button
      class="button numbers-and-extras number0"
      type="button"
    >
      0
    </button>
    <button
      class="button numbers-and-extras"
      type="button"
    >
      .
    </button>
    <button
      class="button math-symbols"
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
